src/lib/system_logger.ts

```diff
@@ -0,0 +1,115 @@
+const systemLogTag = '__nfSystemLog'
+
+const serializeError = (error: Error): Record<string, unknown> => {
+  const cause = error?.cause instanceof Error ? serializeError(error.cause) : error.cause
+
+  return {
+    error: error.message,
+    error_cause: cause,
+    error_stack: error.stack,
+  }
+}
+
+// eslint pretends there's a different enum at the same place - it's wrong!
+// eslint-disable-next-line no-shadow
+export enum LogLevel {
+  Debug = 1,
```

Comment:
The edge functions logger has a simpler "Debug/Log/Error" hierarchy here. Since log levels were introduced there only very recently, we can still change what's being used here. I picked those levels because they match the Go levels. @eduardoboucas do you see any value in having an additional `Warn` level and using the `Info` name? Should we change this to be `Debug/Log/Error` as well?

Reply:
No strong opinions. I personally don't feel the need for extra levels, and it's not clear to me when we'd use `warn` vs. `info`.

If we do it, do we make `log()` an alias of `info()`?

Reply:
Then let's stick with Debug/Log/Error for now - we can always add the additional levels in the future.

> If we do it, do we make log() an alias of info()?

That's the current impl, yes.

Reply:
Oops, missed that part!